helpers/legacyStore.js

```diff
@@ -6,13 +6,31 @@ var assign = require("can-util/js/assign/assign");
 
 module.exports = function (count, make, filter) {
 	/*jshint eqeqeq:false */
-	var getUniqueId = (function () {
+	var nextUniqueId = (function () {
 		var i = 0;
 		return function () {
 			return i++;
 		}
 	})();
 
+	// Check to see if an existing item has the ID we want to assign
+	var isIdUnique = function (items, id) {
+		for (var key in items) {
+			if (items[key].id === id) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	var getUniqueId = function (items) {
+		var id = nextUniqueId();
```

Comment:
I think we might be able to calculate the right number only once and cache it.  I think this is `n^2`.